.github/workflows/ci.yaml

```diff
@@ -45,8 +45,20 @@ jobs:
           - run-cargo-clippy
 
     steps:
+      - name: Get commits count
+        id: commits_count
+        run: |
+          echo "::set-output name=commits_count::`curl --silent ${{ github.event.pull_request.commits_url }} | jq length`"
+      - name: Compute commits fetch count
+        id: fetch_count
+        # Fetch an extra commit to make sure that `git diff` with the given number of commits works.
+        run: |
+          echo "::set-output name=commits_fetch_count::$((${{ steps.commits_count.outputs.commits_count }}+1))"
```

Comment:
Maybe these should be combined and produce a single --scope= flag value, which gets used below once and for all (instead of duplicating that step changing only that flag)? i.e. the logic to decide whether we are in pull_request or push would also be moved here

Reply:
I don't think that these can be combined. We still need two numbers, one to fetch the commit, the other to use in the scope flag. But the value of scope flag could be created here. I might leave that for another PR though.

Reply:
Actually looks like the way the checkout action fetches the repository adds a single commit that contains all the changes. So, fetching with depth 2, and setting `--scope=commits:1` gives the list of all modified files in the PR.